SPM Example/PortalSwift/ViewController.swift

```diff
@@ -220,29 +230,151 @@ class ViewController: UIViewController, UITextFieldDelegate {
       throw PortalExampleAppError.userNotLoggedIn()
     }
     guard let config else {
-      self.logger.error("ViewController.recover() - ❌ Application configuration not set.")
+      self.logger.error("ViewController.eject() - ❌ Application configuration not set.")
       throw PortalExampleAppError.configurationNotSet()
     }
-    guard let cipherTextUrl = URL(
-      string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/cipher-text/fetch?backupMethod=\(withBackupMethod.rawValue)"
-    ) else {
-      throw URLError(.badURL)
+
+    var cipherText: String? = nil
+    var organizationShare: String? = nil
+
+    guard let client = try await portal.client else {
+      throw PortalExampleAppError.clientInformationUnavailable("No client found.")
     }
-    let cipherTextData = try await requests.get(cipherTextUrl)
-    let cipherTextResponse = try decoder.decode(CipherTextResult.self, from: cipherTextData)
 
-    guard let organizationBackupShareUrl = URL(
-      string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/org-share/fetch?backupMethod=\(withBackupMethod.rawValue)-SECP256K1"
-    ) else {
-      throw URLError(.badURL)
+    let backupWithPortal = client.environment?.backupWithPortalEnabled ?? false
+
+    if !backupWithPortal {
+      guard let cipherTextUrl = URL(
+        string: "\(config.custodianServerUrl)/mobile/\(user.exchangeUserId)/cipher-text/fetch?backupMethod=\(withBackupMethod.rawValue)"
+      ) else {
+        throw URLError(.badURL)
+      }
+      let cipherTextData = try await requests.get(cipherTextUrl)
+      let cipherTextResponse = try decoder.decode(CipherTextResult.self, from: cipherTextData)
+
+      cipherText = cipherTextResponse.cipherText
```

Comment:
```suggestion
      cipherText = try await getCipherText(custodianServerUrl: config.custodianServerUrl, exchangeUserId: user.exchangeUserId, method: withBackupMethod).cipherText
```

Moving that logic to a separate private function and just consume it here should make this code more readable